### ISSUE TITLE
plugin-regexp/remove-useless-group should not remove groups in `.split()` or `.replace()`

### ISSUE THREAD

**Issue by egilll** [coderaiser/putout#74]
[plugin-regexp/remove-useless-group](https://github.com/coderaiser/putout/tree/master/packages/plugin-regexp/lib/remove-useless-group) changes:
```js
'lala'.split(/(a)/g); // Output: ["l", "a", "l", "a", ""]
```
into: 
```js
'lala'.split(/a/g); // Output: ["l", "l", ""]
```

It also changes:
```js
'lala'.replace(/(a)/g,'-$1-'); // Output: "l-a-l-a-"
```
into:
```js
'lala'.replace(/a/g,'-$1-'); // Output: "l-$1-l-$1-"
```

This plugin is safe to use in `.test()`, but should likely ignore regex that appears in any other function (match, split, replace).

**Comment by coderaiser** [comment]
Thanks a lot for such a detailed examples :)! Landed in `@putout/plugin-regexp@2.6.0` 🎉 . Is it works for you?

**Comment by egilll** [comment]
Wonderful, now works correctly when inside of a split or replace.
There are three other situations I've been able to find where this results in changing the output:

1. Standalone regex

```js
// Before
const r = /(a)/g;
"lala".split(r);  // Output: ["l", "a", "l", "a", ""]
```

```js
// After
const r = /a/g;
"lala".split(r); // Output: ["l", "l", ""]
```

2. exec()

```js
// Before
/(a)/g.exec("haha"); // Output: ["a", "a"]
```
```js
// After
/a/g.exec("haha"); // Output: ["a"]
```

3. replaceAll()

```js
// Before
"lala".replaceAll(/(a)/g,'-$1-'); // Output: "l-a-l-a-"
```
```js
// After
"lala".replaceAll(/a/g, "-$1-"); // Output: "l-$1-l-$1-"
```

A possible solution (in particular to the standalone regex) might be to whitelist regex that is:
- Directly followed by .test()
- Directly followed by .search()
- Inside of .match()

instead of blacklisting?

(By the way, this is a *great* project that I've been much in need of!)

**Comment by coderaiser** [comment]
Landed in `@putout/plugin-regexp@v2.7.0`

> (By the way, this is a great project that I've been much in need of!)

Glad to read and always ready to help :). Is it works for you?

**Comment by egilll** [comment]
Still fails on variables other than `const`, and standalone RegEx objects:

```js
var x = /(a)/g;
let y = /(a)/g;
const z = (() => {
  return /(a)/g;
})();
```

The benefit of this transformation may be too minor relative to the inconvenience it could cause.

**Comment by coderaiser** [comment]
Do you want [to help me with this](https://github.com/coderaiser/putout/blob/master/packages/plugin-regexp/lib/remove-useless-group/index.js) :)?

**Comment by coderaiser** [comment]
Landed in `@putout/plugin-regexp@2.8.0` 🎉 . Is it works for you?
